core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -138,18 +138,8 @@ sealed abstract class IO[+E, +A] extends Serializable { self =>
    *
    * TODO: Replace with optimized primitive.
    */
-  final def parWith[E1 >: E, B, C](that: IO[E1, B])(f: (A, B) => C): IO[E1, C] = {
-    val s: IO[Nothing, Either[E1, A]] = self.attempt
-    s.raceWith[E1, Either[E1, A], Either[E1, B], C](that.attempt)(
-      {
-        case (Left(e), fiberb)  => fiberb.interrupt *> IO.fail[E1](e)
-        case (Right(a), fiberb) => IO.absolve[E1, B](fiberb.join).map((b: B) => f(a, b))
-      }, {
-        case (Left(e), fibera)  => fibera.interrupt *> IO.fail[E1](e)
-        case (Right(b), fibera) => IO.absolve[E1, A](fibera.join).map((a: A) => f(a, b))
-      }
-    )
-  }
+  final def parWith[E1 >: E, B, C](that: IO[E1, B])(f: (A, B) => C): IO[E1, C] =
```

Comment:
In the event one side fails, `parWith` needs to interrupt the other side, so the minimum amount of computation is performed. That's why it's old implementation is so complex—this simplification is very elegant but unfortunately destroys that property.

Reply:
If I supervise this action, then any failure will make the whole action fail, which will lead to `catchError` collecting the finalizers and run them, thereby killing all fibers. If this is incorrect _or_ supervision kills performance, I have another version that uses `parWith` and observes the fibers.

Reply:
Such a conundrum 😅 The `raceWith` version looks as elegant imho, but I like the idea of having a separate `Fiber#parWith` (or `zipWithPar`, according to the decided terminology?)